src/components/PostDetails.tsx

```diff
@@ -2,105 +1,109 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import * as commentsService from '../api/comments';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Comment as CommentItem } from '../types/Comment';
+import { Post } from '../types/Post';
+import Comment from './Comment';
+
+type Props = {
+  postSelected: Post;
+};
+export const PostDetails: React.FC<Props> = ({ postSelected }) => {
+  const [comments, setComments] = useState<CommentItem[]>([]);
+
+  const [isError, setIsError] = useState<boolean>(false);
+  const [isLoading, setIsLoadind] = useState(false);
+  const [isAddingComment, setIsAddingComment] = useState<boolean>(false);
+
+  useEffect(() => {
+    setIsLoadind(true);
+
+    commentsService
+      .getComments(postSelected.id)
+      .then(setComments)
+      .catch(() => {
+        setIsError(true);
+      })
+      .finally(() => setIsLoadind(false));
+  }, [postSelected]);
+
+  useEffect(() => {
+    setIsAddingComment(false);
+  }, [postSelected]);
+
+  const addNewComment = useCallback((newComment: CommentItem) => {
+    setComments(prevComments => [...prevComments, newComment]);
+  }, []);
+
+  const handleDeleteComment = useCallback((commentId: number) => {
+    try {
+      setComments(prev => prev.filter(comment => comment.id !== commentId));
+      commentsService.deleteComment(commentId);
+    } catch (error) {
+      setIsError(true);
+    }
+  }, []);
 
-export const PostDetails: React.FC = () => {
   return (
     <div className="content" data-cy="PostDetails">
       <div className="content" data-cy="PostDetails">
```

Comment:
Duplicate `data-cy="PostDetails"` attribute. This is unnecessary and should be removed to avoid confusion.